dvc/fs/http.py

```diff
@@ -127,3 +129,14 @@ def fs(self):
 
     def unstrip_protocol(self, path: str) -> str:
         return path
+
+    def put_file(
+        self,
+        from_file: AnyFSPath,
+        to_info: AnyFSPath,
+        callback: Any = DEFAULT_CALLBACK,
+        **kwargs,
+    ) -> None:
+        kwargs["method"] = self.upload_method
```

Comment:
Let's use a `setdefault` here?

```suggestion
        kwargs.setdefault("method", self.upload_method)
```